packages/trader/src/Stores/Modules/Trading/trade-store.js

```diff
@@ -682,17 +689,23 @@ export default class TradeStore extends BaseStore {
         if (!proposal_info) {
             return;
         }
-        const { contract_type, barrier, high_barrier, low_barrier } = proposal_info;
+        const { barrier, contract_type, high_barrier, low_barrier } = proposal_info;
 
         if (isBarrierSupported(contract_type)) {
             const color = this.root_store.ui.is_dark_mode_on ? BARRIER_COLORS.DARK_GRAY : BARRIER_COLORS.GRAY;
-            const turbos_color = getTurbosColor(contract_type);
+            const turbos_hovered_color = contract_type === 'TURBOSSHORT' ? BARRIER_COLORS.RED : BARRIER_COLORS.GREEN;
 
             // create barrier only when it's available in response
-            this.main_barrier = new ChartBarrierStore(barrier || high_barrier, low_barrier, this.onChartBarrierChange, {
-                color: isTurbosContract(contract_type) ? turbos_color : color,
-                not_draggable: isTurbosContract(contract_type),
-            });
+            this.main_barrier = new ChartBarrierStore(
+                this.hovered_barrier || barrier || high_barrier,
+                low_barrier,
+                this.onChartBarrierChange,
+                {
+                    color: isTurbosContract(contract_type) && this.hovered_barrier ? turbos_hovered_color : color,
```

Comment:
Maybe we can remove `isTurbosContract(contract_type)` because only turbos has `this.hovered_barrier`?

Reply:
@kate-deriv yeah, we surely can). let's also rename the variable:
 `turbos_hovered_color` => `hovered_barrier_color`
 So that we can reuse this logics later, not only for Turbos, if needed

Reply:
Sure, thank you